pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/UnnecessaryImport.xml

```diff
@@ -1474,8 +1474,281 @@ public class Foo {
+
+<!--                public void handleConfigChange(Map<String, ConfigValue> buildTimeRuntimeValues) {-->
+<!--                    if (!mismatches.isEmpty()) {-->
+<!--                        var msg = "Build time property cannot be changed at runtime:\n" + String.join("\n", mismatches);-->
+<!--                        var buildTimeMismatchAtRuntime = config-->
+<!--                                .getOptionalValue("quarkus.config.build-time-mismatch-at-runtime", String.class)-->
+<!--                                .orElse(warn);-->
+<!--                        if (fail.equals(buildTimeMismatchAtRuntime)) {-->
+<!--                            throw new IllegalStateException(msg);-->
+<!--                        } else if (warn.equals(buildTimeMismatchAtRuntime)) {-->
+<!--                            log.warn(msg);-->
+<!--                        }-->
+<!--                    }-->
+<!--                }-->
+<!--            }-->
+<!--        ]]></code>-->
+<!--    </test-code>-->
+
+
+    <!-- Test cases for static imports of nested classes -->
+    <test-code>
+        <description>Static import of nested class</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import static java.util.Map.Entry;
+public class Foo {
+    Entry<String, String> entry;
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Static on-demand import of nested class</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import static java.util.Map.*;
+public class Foo {
+    Entry<String, String> entry;
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in annotations -->
+    <test-code>
+        <description>Import used in annotation value</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import some.pkg.MyEnum;
+@MyAnnotation(MyEnum.VALUE)
+public class Foo {}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Import used in array annotation value</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import some.pkg.MyEnum;
+@MyAnnotation({MyEnum.VALUE1, MyEnum.VALUE2})
+public class Foo {}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in lambda parameters -->
+    <test-code>
+        <description>Import used in lambda parameter type</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.io.File;
+public class Foo {
+    void test() {
+        list.stream().map((File f) -> f.getName());
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <!-- Test cases for imports used in method references -->
+    <test-code>
+        <description>Import used in method reference</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+import java.io.File;
+public class Foo {
+    void test() {
+        list.stream().map(File::getName);
```

Comment:
this won't compile: `list` is not known here, so PMD won't know, what the method `stream()` will return etc...